common/src/api/external/mod.rs

```diff
@@ -681,6 +681,7 @@ impl TryFrom<i64> for Generation {
 #[display(style = "kebab-case")]
 pub enum ResourceType {
     AddressLot,
+    AddressLotBlock,
```

Comment:
I added `AddressLotBlock` as a dedicated resource because it's represented distinctly in the API but I do wonder if it really should be a resource. `AddressLot` by itself (as I [mentioned in the parent PR](https://github.com/oxidecomputer/omicron/pull/2645/files#r1164898672)) feels incomplete without a `blocks` property. I get that adding that wouldn't work if we wanted it to be a paginated resource (which I assume is the case).

Reply:
Logically I agree that `AddressLot` and `AddressLotBlock` are incomplete without each other. The blocks are broken out because there is the potential for the number to be quite large.